developers/release.md

```diff
@@ -0,0 +1,214 @@
+
+### Step 1: Warn Community of Branching
+
+The first step to any release is warning the community of impending branching prior to the branch date to give them time to push on any pull requests that are pending. The email should be sent no less than 2 weeks prior to the intended branch date and ideally when Foreman sends their branching announcement.
+
+### Step 2: Stable Pulp Build
+
+Prior to branching the Katello repositories, the release nanny should figure out what the current stable version of Pulp is and if not already tagged and built, do so. Please see the [Upgrading Pulp Guide](http://www.katello.org/developers/upgrade_pulp.html).
+
+### Step 3: Setup Your Environment
+
+There are a number of repositories that will need to be tagged and branched throughout the process. We recommend organizing this before hand so that you have a common place to reference and work through these tasks within throughout the release process. Clone the following:
+
+```
+git clone https://github.com/Katello/katello
+git clone https://github.com/Katello/katello-installer.git
+git clone https://github.com/Katello/katello-misc
+git clone https://github.com/Katello/katello-agent
+git clone https://github.com/Katello/hammer-cli-katello
+```
+
+Note that some repositories contain multiple spec-files and thus packages:
+
+```
+katello
+  rubygem-katello.spec
+  deploy/katello.spec
+
+katello-misc
+  certs-tools/katello-certs-tools.spec
+  repos/katello-repos.spec
+  utils/katello-utils.spec
+
+katello-agent
+  katello-agent.spec
+
+katello-installer
+  katello-installer.spec
+
+hammer-cli-katello
+  rubygem-hammer_cli_katello.spec
+```
+
+### Step 4: Configure Koji
+
+In preparation for the build, Koji needs to have configuration added specific to the tags that the release will occur under. The first step is to add new build roots and tags:
+
+```
+wave hands, perform magic
+```
+
+### Step 5: Setup Mash Scripts
+
+In order to generate the release repositories on Koji, a mash script for the version being released needs to be added. If you do not have direct access to the koji.katello.org box, you can request access or ask someone who does to perform this step for you.
+
+Copy previous mash script to new one
+
+```
+cp /usr/local/bin/katello-mash-split-2.0.py /usr/local/bin/katello-mash-split-2.1.py
+```
+
+Open new mash script and replace the version with the new one
+
+```
+vim /usr/local/bin/katello-mash-split-2.1.py
+:%s/2.0/2.1/g
+```
+
+Now do the same for all of the 2.0 mash files in `/etc/mash`, copying them to 2.1 files and updating the contents from 2.0 to 2.1. Finally, generate the first set of X.Y repositories on koji:
+
+```
+/usr/local/bin/katello-mash-split-2.1.py
+```
+
+### Step 6: Get Your Branch On
+
+On the date and time you previously warned the community branching would occur on, do the following for each repository EXCEPT `hammer-cli-katelo`.
```

Comment:
you might mention what to do for hammer-cli-katello, which is:

1.  if there are no changes since last version do nothing
2. if there are changes, bump version, release the gem, tag, build in koji